commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/portlet/action/EditCommerceAccountUserMVCActionCommand.java

```diff
@@ -218,92 +228,155 @@ protected void updatePassword(
+				login = String.valueOf(user.getUserId());
 			}
 
-			_userLocalService.updatePasswordReset(
-				user.getUserId(), passwordReset);
+			_authenticatedSessionManager.login(
+				_portal.getOriginalServletRequest(
+					_portal.getHttpServletRequest(actionRequest)),
+				_portal.getHttpServletResponse(actionResponse), login,
+				newPassword1, false, null);
+		}
+	}
 
-			if (Validator.isNotNull(reminderQueryQuestion) &&
-				Validator.isNotNull(reminderQueryAnswer)) {
+	protected void updateUser(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		long commerceAccountId = ParamUtil.getLong(
+			actionRequest, "commerceAccountId");
+
+		try {
+			long userId = ParamUtil.getLong(actionRequest, "userId");
 
-				_userLocalService.updateReminderQuery(
-					user.getUserId(), reminderQueryQuestion,
-					reminderQueryAnswer);
+			if (themeDisplay.getUserId() != userId) {
+				_commerceAccountPermission.check(
+					themeDisplay.getPermissionChecker(), commerceAccountId,
+					CommerceAccountActionKeys.MANAGE_MEMBERS);
 			}
 
-			if ((user.getUserId() == themeDisplay.getUserId()) &&
-				passwordModified) {
+			String screenName = ParamUtil.getString(
+				actionRequest, "screenName");
+			String emailAddress = ParamUtil.getString(
+				actionRequest, "emailAddress");
+			String firstName = ParamUtil.getString(actionRequest, "firstName");
+			String lastName = ParamUtil.getString(actionRequest, "lastName");
 
-				String login = null;
+			boolean deleteLogo = ParamUtil.getBoolean(
+				actionRequest, "deleteLogo");
 
-				Company company = themeDisplay.getCompany();
+			byte[] portraitBytes = null;
 
-				String authType = company.getAuthType();
+			long fileEntryId = ParamUtil.getLong(actionRequest, "fileEntryId");
 
-				if (authType.equals(CompanyConstants.AUTH_TYPE_EA)) {
-					login = user.getEmailAddress();
-				}
-				else if (authType.equals(CompanyConstants.AUTH_TYPE_SN)) {
-					login = user.getScreenName();
-				}
-				else if (authType.equals(CompanyConstants.AUTH_TYPE_ID)) {
-					login = String.valueOf(user.getUserId());
-				}
+			if (fileEntryId > 0) {
+				FileEntry fileEntry = _dlAppLocalService.getFileEntry(
+					fileEntryId);
 
-				_authenticatedSessionManager.login(
-					_portal.getOriginalServletRequest(
-						_portal.getHttpServletRequest(actionRequest)),
-					_portal.getHttpServletResponse(actionResponse), login,
-					newPassword1, false, null);
+				portraitBytes = FileUtil.getBytes(fileEntry.getContentStream());
 			}
+
+			ServiceContext serviceContext = ServiceContextFactory.getInstance(
+				User.class.getName(), actionRequest);
+
+			User user = _userLocalService.getUser(userId);
+
+			Date birthday = user.getBirthday();
+
+			Calendar birthdayCal = CalendarFactoryUtil.getCalendar(
+				birthday.getTime());
+
+			// Update user
+
+			_userLocalService.updateUser(
```

Comment:
Wrap this method in the AccountUSerService and inside that check for permisison to edit th eaccount and manage members

Reply:
See the updated file plz